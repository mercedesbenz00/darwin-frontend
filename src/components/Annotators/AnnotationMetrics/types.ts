import { MembershipPayload, DatasetAnnotationReportPayload } from '@/store/types'

export type AnnotationDataRange = 'total' | 'month' | 'week' | 'day'
export type AnnotationDataGranularity = 'day' | 'hour'

export type ChartDimension =
  'annotationTime' |
  'imagesAnnotated' |
  'avgTimePerAnnotation' |
  'annotationsCreated' |
  'reviewPassRate'

export type AdaptedDataPoint =
  {[k in Exclude<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number } &
  {[k in Extract<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number | null} &
  { userId: number | null, date: string }

export type AggregatedDataPoint = Pick<
    DatasetAnnotationReportPayload,
    | 'annotation_time'
    | 'annotations_approved'
    | 'annotations_created'
    | 'date'
    | 'images_annotated'
    | 'images_approved'
    | 'images_rejected'
    | 'user_id'
  >

export type AnnotationDimensionData = {
  annotationTime: number,
  annotationsCreated: number,
  avgTimePerAnnotation: number | null,
  imagesAnnotated: number,
  reviewPassRate: number | null
}

export type AnnotationChartData = {
  data: AdaptedDataPoint[]
  dimensions: AnnotationDimensionData
  member?: MembershipPayload
  visible: boolean
}

export type AnnotationBarChartData = {
  data: AggregatedDataPoint[]
  member?: MembershipPayload
}

/**
 * Format for a single item to be shown in a chart tooltip.
 *
 * A tooltip will contain multiple items
 */
export type ChartTooltipData = { member: MembershipPayload | null, value: string}
