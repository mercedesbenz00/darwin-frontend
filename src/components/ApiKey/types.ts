export type AbilityGroup =
  'manage_comments' |
  'manage_tasks' |
  'manage_datasets' |
  'manage_team' |
  'manage_team_members' |
  'manage_models'

export type Group = { id: AbilityGroup; name: string }
export type GroupOption = Group & { selected: boolean }

export enum SupportedLanguages {
  shell = 'shell',
  python = 'python',
  javascript = 'javascript',
  elixir = 'elixir',
  cli = 'cli'
}

export type AccessCodeExample = { language: SupportedLanguages, code: string }
