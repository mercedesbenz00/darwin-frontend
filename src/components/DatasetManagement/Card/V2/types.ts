import {
  DatasetItemPayload,
  V2DatasetItemPayload,
  V2DatasetFolderPayload
} from '@/store/types'

export type DatasetFolderCardProps = {
  data: V2DatasetFolderPayload
  readonly: boolean
  urlPrefix: string | null
}

export type DatasetItemCardProps = {
  data: DatasetItemPayload,
  dataV2: V2DatasetItemPayload
}
