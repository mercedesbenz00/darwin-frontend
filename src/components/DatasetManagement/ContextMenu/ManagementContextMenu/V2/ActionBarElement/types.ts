export enum ActionBarType {
  DEFAULT = 'default',
  SELECT = 'select',
  DANGER = 'danger'
}

export type ActionBarElement = {
  active?: boolean
  disabled?: string
}

export enum DangerMode {
  ARCHIVE = 'archive',
  DELETE = 'delete'
}
