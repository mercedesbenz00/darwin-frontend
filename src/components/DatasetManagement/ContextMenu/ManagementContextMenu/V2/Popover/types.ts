export enum PopoverContextType {
  ASSIGN = 'assign',
  STAGE = 'stage',
  FOLDER = 'folder',
  PRIORITY = 'priority'
}
