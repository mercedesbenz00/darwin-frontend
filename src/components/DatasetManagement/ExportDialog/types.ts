import { DropdownOption } from '@/components/Common/Dropdown/types'

export type ExportFormatPillType = {
  name: string;
  type: 'feather-light' | 'pink' | 'yellow';
}

export type ExportFormatOptionType = DropdownOption & {
  id: string
  label: string
  pills?: ExportFormatPillType[]
}
