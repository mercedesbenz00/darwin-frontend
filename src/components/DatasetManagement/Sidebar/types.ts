import { DatasetItemStatus } from '@/store/types'

export type ClassFilterChange = {
  positiveOptions: number[]
  negativeOptions: number[]
}

export type StatusFilterChange = {
  commented: boolean
  positiveOptions: DatasetItemStatus[]
  negativeOptions: DatasetItemStatus[]
}

export type GenericFilterChange = {
  positiveAssignees: number[]
  negativeAssignees: number[]
  positiveFilenames: string[]
  negativeFilenames: string[]
  positivePaths: string[]
  negativePaths: string[]
}
