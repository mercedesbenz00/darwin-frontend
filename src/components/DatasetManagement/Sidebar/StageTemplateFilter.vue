<template>
  <stage-template-filter
    v-if="activeWorkflowTemplate"
    :positive-stage-template-ids="positiveStageTemplateIds"
    :negative-stage-template-ids="negativeStageTemplateIds"
    :workflow-template="activeWorkflowTemplate"
    @change="updateFilter"
  />
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import { State } from 'vuex-class'

import { StageTemplateFilter } from '@/components/DatasetFiltering/StageTemplateFilter'
import { DatasetPayload, RootState, WorkflowTemplatePayload } from '@/store/types'

@Component({
  name: 'workflow-sidebar-stage-template-filter',
  components: { StageTemplateFilter }
})
export default class WorkflowSidebarStageTemplateFilter extends Vue {
  @Prop({ required: true, type: Object })
  dataset!: DatasetPayload

  @Prop({ required: false, default: () => [] })
  positiveStageTemplateIds!: number[]

  @Prop({ required: false, default: () => [] })
  negativeStageTemplateIds!: number[]

  @State((state: RootState) => state.dataset.workflowTemplates)
  workflowTemplates!: WorkflowTemplatePayload[]

  get activeWorkflowTemplate (): WorkflowTemplatePayload | null {
    return this.workflowTemplates.find(
      t => t.id === this.dataset.default_workflow_template_id
    ) || null
  }

  updateFilter (
    params: {
      positiveStageTemplateIds: number[],
      negativeStageTemplateIds: number[]
    }
  ): void {
    this.$emit('update:positive-stage-template-ids', params.positiveStageTemplateIds)
    this.$emit('update:negative-stage-template-ids', params.negativeStageTemplateIds)
    this.$emit('change', params)
  }
}
</script>
