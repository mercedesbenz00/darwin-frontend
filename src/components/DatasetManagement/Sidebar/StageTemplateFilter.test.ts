import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'

import createTestStore from 'test/unit/createTestStore'
import { buildDatasetPayload } from 'test/unit/factories'
import { initializeARTemplate } from 'test/unit/factories/helpers'
import { emitRootStub, nthEmitted, rootStubProps } from 'test/unit/testHelpers'

import StageTemplateFilter from '@/components/DatasetManagement/Sidebar/StageTemplateFilter.vue'
import { DatasetPayload } from '@/store/types'

const localVue = createLocalVue()
localVue.use(Vuex)

let propsData: {
  dataset: DatasetPayload
  positiveStageTemplateIds?: number[]
}
let store: ReturnType<typeof createTestStore>

const sfh = buildDatasetPayload({ id: 5, default_workflow_template_id: 9 })
const template = initializeARTemplate(sfh)
template.id = 9

beforeEach(() => {
  propsData = {
    dataset: sfh,
    positiveStageTemplateIds: []
  }
  store = createTestStore()
  store.commit('dataset/SET_CURRENT_DATASET_DETAILS', { id: 5 })
  store.commit('dataset/PUSH_WORKFLOW_TEMPLATE', template)
})

it('matches snapshot', () => {
  const wrapper = shallowMount(StageTemplateFilter, { localVue, propsData, store })
  expect(wrapper).toMatchSnapshot()
})

it('sets correct props on child', () => {
  const wrapper = shallowMount(StageTemplateFilter, { localVue, propsData, store })
  expect(rootStubProps(wrapper, 'workflowTemplate')).toEqual(template)
})

it('adds positive stage template to filter', async () => {
  const wrapper = shallowMount(StageTemplateFilter, { localVue, propsData, store })
  await emitRootStub(wrapper, 'change', { positiveStageTemplateIds: [5] })
  expect(nthEmitted(wrapper, 'update:positive-stage-template-ids', 0)).toEqual([[5]])
  expect(nthEmitted(wrapper, 'change', 0)).toEqual([{ positiveStageTemplateIds: [5] }])

  await emitRootStub(wrapper, 'change', { positiveStageTemplateIds: [] })
  expect(nthEmitted(wrapper, 'update:positive-stage-template-ids', 1)).toEqual([[]])
  expect(nthEmitted(wrapper, 'change', 1)).toEqual([{ positiveStageTemplateIds: [] }])
})

it('adds negative stage template to filter', async () => {
  const wrapper = shallowMount(StageTemplateFilter, { localVue, propsData, store })
  await emitRootStub(wrapper, 'change', { negativeStageTemplateIds: [5] })
  expect(nthEmitted(wrapper, 'update:negative-stage-template-ids', 0)).toEqual([[5]])
  expect(nthEmitted(wrapper, 'change', 0)).toEqual([{ negativeStageTemplateIds: [5] }])

  await emitRootStub(wrapper, 'change', { negativeStageTemplateIds: [] })
  expect(nthEmitted(wrapper, 'update:negative-stage-template-ids', 1)).toEqual([[]])
  expect(nthEmitted(wrapper, 'change', 1)).toEqual([{ negativeStageTemplateIds: [] }])
})
