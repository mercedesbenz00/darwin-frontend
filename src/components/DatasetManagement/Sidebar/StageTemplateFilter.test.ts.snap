// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `<stage-template-filter-stub positivestagetemplateids="" negativestagetemplateids="" workflowtemplate="[object Object]"></stage-template-filter-stub>`;
